controllers/storagecluster/storagecluster_controller.go

```diff
@@ -199,6 +207,11 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				},
 			},
 			enqueueStorageClusterRequest,
+		).
+		Watches(
+			&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: onboardingTicketPublicKeySecretName}},
```

Comment:
you don't generally establish watch on specific secret but on secret resource and filter by predicate, so remove the ObjectMeta here